modules/encoders/php/base64.rb

```diff
@@ -8,24 +8,32 @@ class MetasploitModule < Msf::Encoder
 
   def initialize
     super(
-      'Name'             => 'PHP Base64 Encoder',
-      'Description'      => %q{
+      'Name' => 'PHP Base64 Encoder',
+      'Description' => %q{
         This encoder returns a base64 string encapsulated in
         eval(base64_decode()), increasing the size by a bit more than
         one third.
       },
-      'Author'           => 'egypt',
-      'License'          => BSD_LICENSE,
-      'Arch'             => ARCH_PHP)
+      'Author' => 'egypt',
+      'License' => BSD_LICENSE,
+      'Arch' => ARCH_PHP)
   end
 
   def encode_block(state, buf)
     # Have to have these for the decoder stub, so if they're not available,
     # there's nothing we can do here.
-    %w{c h r ( ) . e v a l b a s e 6 4 _ d e c o d e ;}.uniq.each do |c|
+    %w[c h r ( ) . e v a l b a s e 6 4 _ d e c o d e ;].uniq.each do |c|
       raise BadcharError if state.badchars.include?(c)
     end
 
+    # Modern versions of PHP choke on unquoted litteral strings.
+    quote = "'"
+    if state.badchars.include?("'")
+      raise BadcharError if state.badchars.include?('"')
```

Comment:
I think we should be raising a message with this similar to what is being done at https://github.com/rapid7/metasploit-framework/blob/96c9f60cc15ed169ab2689755b2c244e9b67d4dc/modules/encoders/mipsbe/byte_xori.rb#L131-L133?

Reply:
Fixed in 975de9d479554da44118eb356d0a9cf2d23c64a6